src/common/decorators/validation.ts

```diff
@@ -166,3 +169,66 @@ export function IsTxTagOrModuleName(validationOptions?: ValidationOptions) {
     });
   };
 }
+
+@ValidatorConstraint({ async: false })
+export class IsNotSiblingOfConstraint implements ValidatorConstraintInterface {
+  validate(value: unknown, args: ValidationArguments) {
+    if (value !== undefined && value !== null) {
+      return this.getFailedConstraints(args).length === 0;
+    }
+    return true;
+  }
+
+  defaultMessage(args: ValidationArguments) {
+    return `${
+      args.property
+    } cannot exist alongside the following defined properties: ${this.getFailedConstraints(
+      args
+    ).join(', ')}`;
+  }
+
+  getFailedConstraints(args: ValidationArguments) {
+    return args.constraints.filter(prop => {
+      const value = args.object[prop as keyof typeof args.object];
+      return value !== undefined && value !== null;
+    });
+  }
+}
+
+// Create Decorator for the constraint that was just created
+export function IsNotSiblingOf(props: string[], validationOptions?: ValidationOptions) {
+  return function (object: object, propertyName: string) {
+    registerDecorator({
+      target: object.constructor,
+      propertyName,
+      options: validationOptions,
+      constraints: props,
+      validator: IsNotSiblingOfConstraint,
+    });
+  };
+}
+
+// Helper function for determining if a prop should be validated
+export function incompatibleSiblingsNotPresent(incompatibleSiblings: string[]) {
+  return function (o: object, v: unknown) {
+    return Boolean(
+      v !== undefined || // Validate if prop has value
+        incompatibleSiblings.every(prop => o[prop as keyof typeof o] === undefined) // Validate if all incompatible siblings are not defined
+    );
+  };
+}
+
+export function IncompatibleWith(
```

Comment:
```suggestion
/**
 Ensures certain keys are mutually exclusive
 */
export function IncompatibleWith(
```
This is the only thing that needs to be exported so should get a little comment. Other parts should stay private to this file to prevent auto complete suggesting stuff that is never going to be used.